crates/turbopack-core/src/error.rs

```diff
@@ -0,0 +1,60 @@
+use std::fmt::{Display, Formatter, Result};
+
+pub struct PrettyPrintError<'a>(pub &'a anyhow::Error);
+
+impl<'a> Display for PrettyPrintError<'a> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result {
+        let mut i = 0;
+        let mut has_details = false;
+
+        let descriptions = self
+            .0
+            .chain()
+            .map(|cause| cause.to_string())
+            .collect::<Vec<_>>();
+
+        for description in &descriptions {
+            let hidden = description.starts_with("Execution of ");
+            if !hidden {
+                let header =
+                    description
+                        .split_once('\n')
+                        .map_or(description.as_str(), |(header, _)| {
+                            has_details = true;
+                            header
+                        });
+                match i {
+                    0 => write!(f, "{}", header)?,
+                    1 => write!(f, "\n\nCaused by:\n- {}", header)?,
+                    _ => write!(f, "\n- {}", header)?,
+                }
+                i += 1;
+            } else {
+                has_details = true;
+            }
+        }
+        if has_details {
+            write!(f, "\n\nDeveloper details:")?;
```

Comment:
Nit: I'd rather we had something like:
```suggestion
            write!(f, "\n\nDebug info:")?;
```

Technically, all our users are developers.